include/experimental/__p1673_bits/blas3_matrix_product.hpp

```diff
@@ -1737,9 +1737,12 @@ void hermitian_matrix_left_product(
     for (size_type j = 0; j < C.extent(1); ++j) {
       for (size_type i = 0; i < C.extent(0); ++i) {
         C(i,j) = ElementType_C{};
-        for (size_type k = 0; k < A.extent(1); ++k){
-          ElementType_A aik = i <= k ? impl::conj_if_needed(A(k,i)) : A(i,k);
-          C(i,j) += aik * B(k,j);
+        for (size_type k = 0; k < i; ++k){
+          C(i,j) += A(i,k) * B(k,j);
+        }
+        C(i,j) += impl::real_part(A(i,i)) * B(i,j);
+        for (size_type k = i+1; k < A.extent(0); ++k){
+          C(i,j) += impl::conj_if_needed(A(k,i)) * B(k,j);
         }
```

Comment:
It should be legal for `A.extent(0)` to be zero, meaning that `A` has no diagonal element.  Should we thus consider the following change?

```suggestion
        for (size_type k = 0; k < i; ++k){
          C(i,j) += A(i,k) * B(k,j);
        }
        if (A.extent(0) != 0) {
          C(i,j) += impl::real_part(A(i,i)) * B(i,j);
        }
        for (size_type k = i+1; k < A.extent(0); ++k){
          C(i,j) += impl::conj_if_needed(A(k,i)) * B(k,j);
        }
```

Reply:
Well... `A` is square by precondition. That means `A.extent(0) == A.extent(1) == 0`. And... `C.extent(0) == 0` (via multipliable precondition). We should never find oursevles in that part of the loop in this case -> this is a redundant if condition.

Reply:
That's a good point.  While we should consider adding some assert s in there for the precondition, that's out of scope of the PR.